### PR TITLE
Update approver and maintainer practices

### PR DESCRIPTION
**Pull request by svrnm**

### CODE REVIEW

File: content/en/docs/contributing/development.md
Comment (theletterf) [inline content/en/docs/contributing/development.md:151]:
```suggestion
    that merging should be postponed.
```

**Review by theletterf — APPROVED** [review]

File: content/en/docs/contributing/development.md
Comment (chalin) [inline content/en/docs/contributing/development.md:187]: I'd suggest a rebase update. Merge updates can cause problems for long-running PRs.

File: content/en/docs/contributing/development.md
Comment (chalin) [reply content/en/docs/contributing/development.md:146]: IIRC it happening once or twice in the past couple of years. I can't recall for sure how we (docs folks) were notified, but it was either through Slack or through an issue/PR or issue/PR comment either in this repo or in the SIG repo (with a notification of @open-telemetry/docs-approvers). 

I'm not sure that we need to say how a SIG signals though. Your call.

Comment (svrnm) [reply content/en/docs/contributing/development.md:146]: I left it vague on purpose, I would expect them to write an issue comment or ping us via slack, but other means of communication are also acceptable.

File: content/en/docs/contributing/development.md
Comment (svrnm) [reply content/en/docs/contributing/development.md:187]:
```suggestion
- If the branch is out-of-date, rebase update it via the GitHub UI.
```

**Review by tiffany76 — APPROVED** [review]
Makes sense to me. Thanks!

**Comment by theletterf** [comment]
> Warning: Unknown word (chalin)

😅 I'll fix that and merge.

**Pull request merged**
